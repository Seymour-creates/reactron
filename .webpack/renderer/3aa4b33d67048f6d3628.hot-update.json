{"h":"973fd84ce8468e13f3f0","c":{"main_window":true}}