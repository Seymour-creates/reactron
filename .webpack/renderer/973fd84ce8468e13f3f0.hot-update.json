{"h":"e77e8ae52fd264b48d25","c":{"main_window":true}}