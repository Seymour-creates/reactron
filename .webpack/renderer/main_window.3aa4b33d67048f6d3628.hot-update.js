webpackHotUpdate("main_window",{

/***/ "./src/components/ComponentTree.tsx":
false,

/***/ "./src/components/HeadNode.tsx":
false,

/***/ "./src/components/Header.tsx":
false,

/***/ "./src/components/Node.tsx":
false,

/***/ "./src/components/ReactComponent.tsx":
false,

/***/ "./src/components/RenderedContainer.tsx":
false,

/***/ "./src/components/RenderedPage.tsx":
/*!*****************************************!*\
  !*** ./src/components/RenderedPage.tsx ***!
  \*****************************************/
/*! no static exports found */
/***/ (function(module, exports) {

throw new Error("Module parse failed: Unexpected token (37:4)\nFile was processed with these loaders:\n * ./node_modules/ts-loader/index.js\nYou may need an additional loader to handle the result of these loaders.\n|                             react_1.default.createElement(Visualizer_1.default, null))));\n|     div >\n>     ;\n|     ;\n| }");

/***/ }),

/***/ "./src/components/StateContainer.tsx":
false,

/***/ "./src/components/StateItem.tsx":
false,

/***/ "./src/components/Visualizer.tsx":
false,

/***/ "path":
false

})
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsImZpbGUiOiJtYWluX3dpbmRvdy4zYWE0YjMzZDY3MDQ4ZjZkMzYyOC5ob3QtdXBkYXRlLmpzIiwic291cmNlUm9vdCI6IiJ9