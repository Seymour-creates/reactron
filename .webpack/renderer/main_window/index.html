<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>ReacTron</title>
  </head>
  <div id="root"></div>
</html>

<script src="/main_window/index.js"></script>