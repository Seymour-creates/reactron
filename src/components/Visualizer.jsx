import React from 'react';
import StateContainer from './StateContainer.jsx';
import RenderedContainer from './RenderedContainer.jsx';

// shows the selected React Component and it's props / state / methods

export default function Visualizer() {
  return (
    <div className='visualizer'>
      {/* <StateContainer /> */}
      <RenderedContainer />
    </div>
  );
}
