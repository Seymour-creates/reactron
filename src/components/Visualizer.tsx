import React from 'react';
import StateContainer from './StateContainer'
import RenderedContainer from './RenderedContainer'

export default function Visualizer() {
  return (
    <div>
      <StateContainer />
      <RenderedContainer />
    </div>
  );
}
