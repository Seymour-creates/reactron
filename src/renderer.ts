/**
 * This file will automatically be loaded by webpack and run in the "renderer" context.
 * To learn more about the differences between the "main" and the "renderer" context in
 * Electron, visit:
 *
 * https://electronjs.org/docs/tutorial/application-architecture#main-and-renderer-processes
 *
 * By default, Node.js integration in this file is disabled. When enabling Node.js integration
 * in a renderer process, please be aware of potential security implications. You can read
 * more about security risks here:
 *
 * https://electronjs.org/docs/tutorial/security
 *
 * To enable Node.js integration in this file, open up `main.js` and enable the `nodeIntegration`
 * flag:
 *
 * ```
 *  // Create the browser window.
 *  mainWindow = new BrowserWindow({
 *    width: 800,
 *    height: 600,
 *    webPreferences: {
 *      nodeIntegration: true
 *    }
 *  });
 * ```
 */

// renderer is needed to render files to react for electron
// import files here
import './index.scss';
import './main';

// IPC stands for inter-process communication
// ipcRenderer allows async rendering, listens from main processor
const { remote, ipcRenderer } = require('electron');

// ipcMain, event emitter, allows async main processing, will listen from the renderer

const mainProcess = remote.require('./index');

// selectMe button in RenderedPage
const testVar = document.getElementById('selectMe');
// onClick, populate the page with file contents
testVar.addEventListener('click', () => {
  mainProcess.testFile();
});

// ipcRenderer.on('file-opened', (event, ...args) => {
//   console.log(args)
// })

console.log(
  'ğŸ‘‹ This message is being logged by "renderer.js", included via webpack'
);