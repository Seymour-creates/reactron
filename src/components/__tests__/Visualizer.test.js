import {render, cleanup, screen, fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import Visualizer from '../Visualizer';

afterEach(() => {
  cleanup();
});

test('test', () => {
  expect(true).toBe(true);
});

test('should render Rendered Page', () => {
  render(<Visualizer />);
  const visualizer = screen.getByTestId('Visualizer');
  expect(visualizer).toBeInTheDocument();
  expect(visualizer).toContainHTML('div');
  expect(visualizer).toContainHTML('iframe');
  expect(visualizer).toHaveClass('visualizer');
});

// if snapshot test fails, press u in the test terminal to update snapshot
xtest('should match Visualizer snapshot', () => {
  const snapVis = renderer.create(<Visualizer />).toJSON();
  expect(snapVis).toMatchSnapshot();
});
