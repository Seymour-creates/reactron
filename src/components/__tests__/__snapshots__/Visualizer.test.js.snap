// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Visualizer snapshot 1`] = `
<div
  className="visualizer"
  data-testid="Visualizer"
>
  <div
    className="renderedContainer"
  >
    <div
      className="reactComponent"
    >
      <iframe
        src="http://localhost:5000/"
      />
    </div>
  </div>
</div>
`;
